components/column-linking-table.vue

```diff
@@ -18,23 +19,138 @@
 
 <script>
 
+    // Allows for reference to store data by creating simple, implicit getters
+    // Fields listed in mapState below can be found in the store (index.js)
+    import { mapState } from "vuex";
+
+    // Allows for reference to store actions (index.js)
+    import { mapActions } from "vuex";
+
     export default {
 
         props: {
 
-            categoryClasses: { type: Object, required: true },
-            columnToCategoryMap: { type: Object, required: true },
-            fields: { type: Array, required: true },
-            selectedCategory: { type: String, required: true },
-            tableData: { type: Array, required: true }
+            selectedCategory: { type: String, required: true }
+        },
+
+        data() {
+
+            return {
+
+                columnToCategoryTable: [],
+
+                uiText: {
+
+                    tableFields: [
+
+                        { key: "column" },
+                        { key: "description" }
+                    ]
+                }
+            };
+        },
+
+        created() {
+
+            // Create the data structure for the category to column linking table
+            this.setupColumnToCategoryTable();
+        },
+
+        computed: {
+
+            ...mapState([
+
+                "categoryClasses",
+                "columnToCategoryMap",
+                "dataDictionary",
+                "dataTable"
+            ])
         },
 
         methods: {
 
+            ...mapActions([
+
+                "linkColumnWithCategory",
+                "unlinkColumnFromCategory"
+            ]),
+
             applyCategory(p_row, p_index, p_event) {
 
-                // Tell the parent page that a column has been linked with a category
-                this.$emit("column-name-selected", { column: p_row.column });
+                const dataForStore = { column: p_row.column };
+
+                // 1. Link or unlink the currently-selected category and the clicked column
+                if ( this.selectedCategory !== this.columnToCategoryMap[p_row.column] ) {
+
+                    this.linkColumnWithCategory({...dataForStore, category: this.selectedCategory});
+                } else {
+
+                    this.unlinkColumnFromCategory(dataForStore);
+                }
+
+                // 2. Tell the categorization page a column has been clicked
+                this.$emit("column-name-clicked", dataForStore);
```

Comment:
Why do we need this emit here? At this point I have already told the store about the column click. Could whoever is listening to this emit not receive that information from the store directly?

Reply:
As discussed in our huddle, the current function of this emit is to let the parent page know to check if the next page (annotation page) is ready to be accessed given the state of the store.

I am leaving in the `emit` for now as I would like to handle the general move of next page access to the layout/store in a separate PR.